packages/core/strapi/lib/core-api/service/single-type.js

```diff
@@ -22,22 +22,19 @@ const createSingleTypeService = ({ model, strapi, utils }) => {
      *
      * @return {Promise}
      */
-    async createOrUpdate(params = {}) {
+    async createOrUpdate({ data, ...params } = {}) {
```

Comment:
why don't you pass params as is directly ? the spread looks useless

Reply:
It's to prevent passing data as a parameter to `this.find` as it doesn't belong there.

Reply:
then it should also remove the files property

Reply:
True